Parse/src/main/java/com/parse/ParseEncoder.java

```diff
@@ -120,6 +122,12 @@ public Object encode(Object object) {
         return ((ParseQuery.RelationConstraint) object).encode(this);
       }
 
+      if (object instanceof Pattern) {
+        JSONObject json = new JSONObject();
+        json.put("$regex", object.toString());
```

Comment:
I think that could be better to avoid update the ParseEncoder. I didn't researched a lot.

Now, I think could be better to update the `constraintsAllStartsWith` to:
```
  public ParseQuery<T> whereContainsAllStartsWith(String key, Collection<String> values) {
    checkIfRunning();

    ArrayList<KeyConstraints> startsWithConstraints = new ArrayList<>();
    for (String value : values) {
      KeyConstraints keyConstraints = new KeyConstraints();
      keyConstraints.put("$regex", buildStartsWithRegex(value));
      startsWithConstraints.add(keyConstraints);
    }

    return whereContainsAll(key, startsWithConstraints);
  }
```

This might auto compile as expected. If people prefer to revert the ParseEncoder changes, I'll update the PR 👍